catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java

```diff
@@ -0,0 +1,193 @@
+      BundleContext bundleContext,
+      QueryApplication queryApplication) {
+    this.queryResponseTransformers = queryResponseTransformers;
+    this.bundleContext = bundleContext;
+    this.queryApplication = queryApplication;
+  }
+
+  @Override
+  public Object handle(Request request, Response response) throws Exception {
+    String transformerId = request.params(":transformerId");
+    String body = util.safeGetBody(request);
+    CqlRequest cqlRequest = mapper.readValue(body, CqlRequest.class);
+    CqlQueryResponse cqlQueryResponse = queryApplication.executeCqlQuery(cqlRequest);
+
+    LOGGER.trace("Finding transformer to transform query response.");
+
+    ServiceReference<QueryResponseTransformer> queryResponseTransformer =
+        queryResponseTransformers
+            .stream()
+            .filter(transformer -> transformer.getProperty("id").equals(transformerId))
+            .findFirst()
+            .orElse(null);
+
+    if (queryResponseTransformer == null) {
+      LOGGER.debug("Could not find transformer with id: {}", transformerId);
+      response.status(HttpStatus.NOT_FOUND_404);
+      return mapper.toJson(ImmutableMap.of("message", "Service not found"));
+    }
+
+    attachFileToResponse(request, response, queryResponseTransformer, cqlQueryResponse);
+
+    return "";
+  }
+
+  private void setHttpAttributes(Request request, Response response, BinaryContent content)
+      throws MimeTypeException {
+    String mimeType = content.getMimeTypeValue();
+    String fileExt = getFileExtFromMimeType(mimeType);
+
+    String acceptEncoding = request.headers(HttpHeaders.ACCEPT_ENCODING).toLowerCase();
+
+    if (acceptEncoding.contains(GZIP)) {
+      LOGGER.trace("Request header accepts gzip");
+      response.header(HttpHeaders.CONTENT_ENCODING, GZIP);
+    }
+
+    response.type(mimeType);
+    String attachment =
+        String.format("attachment;filename=export-%s%s", Instant.now().toString(), fileExt);
+    response.header(HttpHeaders.CONTENT_DISPOSITION, attachment);
+  }
+
+  private String getFileExtFromMimeType(String mimeType) throws MimeTypeException {
+    MimeTypes allTypes = MimeTypes.getDefaultMimeTypes();
+    return allTypes.forName(mimeType).getExtension();
+  }
+
+  private void attachFileToResponse(
+      Request request,
+      Response response,
+      ServiceReference<QueryResponseTransformer> queryResponseTransformer,
+      CqlQueryResponse cqlQueryResponse)
+      throws CatalogTransformerException, IOException, MimeTypeException {
+    BinaryContent content =
+        bundleContext
+            .getService(queryResponseTransformer)
+            .transform(cqlQueryResponse.getQueryResponse(), new HashMap<>());
+
+    setHttpAttributes(request, response, content);
+
+    try (OutputStream servletOutputStream = response.raw().getOutputStream();
+        InputStream resultStream = content.getInputStream()) {
+      if (request.headers(HttpHeaders.ACCEPT_ENCODING).toLowerCase().contains(GZIP)) {
```

Comment:
✏️ You could break this condition out into it's own method for readability. It seems like you check for this header in a couple of places too